dlc/src/lib.rs

```diff
@@ -0,0 +1,685 @@
+    local_change_output: TxOut,
+    remote_change_output: TxOut,
+) -> Transaction {
+    let script = make_funding_redeemscript(&local_fund_pubkey, &remote_fund_pubkey);
+
+    let output: Vec<TxOut> = {
+        let outs = vec![
+            TxOut {
+                value: output_amount,
+                script_pubkey: script.to_v0_p2wsh(),
+            },
+            local_change_output,
+            remote_change_output,
+        ];
+        outs.into_iter().filter(|o| o.value >= DUST_LIMIT).collect()
+    };
+
+    let input = [&local_inputs[..], &remote_inputs[..]].concat();
+
+    let funding_transaction = Transaction {
+        version: TX_VERSION,
+        lock_time: 0,
+        input,
+        output,
+    };
+
+    return funding_transaction;
+}
+
+/// Create a refund transaction
+pub fn create_refund_transaction(
+    local_output: TxOut,
+    remote_output: TxOut,
+    funding_input: TxIn,
+    locktime: u32,
+) -> Transaction {
+    Transaction {
+        version: TX_VERSION,
+        lock_time: locktime,
+        input: vec![funding_input],
+        output: vec![local_output, remote_output],
+    }
+}
+
+/// Create the multisig redeem script for the funding output
+pub fn make_funding_redeemscript(a: &PublicKey, b: &PublicKey) -> Script {
+    let our_funding_key = a.to_bytes();
+    let their_funding_key = b.to_bytes();
+
+    Builder::new()
+        .push_opcode(opcodes::all::OP_PUSHNUM_2)
+        .push_slice(&our_funding_key)
+        .push_slice(&their_funding_key)
+        .push_opcode(opcodes::all::OP_PUSHNUM_2)
+        .push_opcode(opcodes::all::OP_CHECKMULTISIG)
+        .into_script()
+}
+
+/// Create an adaptor signature for the given cet
+pub fn create_cet_adaptor_sig<C: secp256k1::Signing>(
+    secp: &secp256k1::Secp256k1<C>,
+    cet: &Transaction,
+    oracle_pubkey: &PublicKey,
+    oracle_r_value: &SchnorrNonce,
+    funding_sk: &PrivateKey,
+    funding_script_pubkey: &Script,
+    total_collateral: u64,
+    msg: &Message,
+) -> Result<(AdaptorSignature, AdaptorProof), Error> {
+    let adaptor_point = secp.schnorr_compute_sig_point(msg, oracle_r_value, &oracle_pubkey.key)?;
+    let sig_hash = util::get_sig_hash_msg(cet, 0, funding_script_pubkey, total_collateral);
+
+    Ok(secp.adaptor_sign(&sig_hash, &funding_sk.key, &adaptor_point))
+}
+
+/// Sign the given ce using own private key, adapt the counter party signature
+/// and place both signatures and the funding multi sig script pubkey on the
+/// witness stack
+pub fn sign_cet<C: secp256k1::Signing>(
+    secp: &secp256k1::Secp256k1<C>,
+    cet: &mut Transaction,
+    adaptor_signature: &AdaptorSignature,
+    oracle_signature: &SchnorrSignature,
+    funding_sk: &PrivateKey,
+    funding_script_pubkey: &Script,
+    total_collateral: u64,
+    is_local: bool,
+) -> Result<(), Error> {
+    let (_, adaptor_secret) = oracle_signature.decompose()?;
+
+    let other_sig = secp.adaptor_adapt(&adaptor_secret, adaptor_signature);
```

Comment:
nit: `complete_sig` might be a nicer name?

Reply:
`complete_full_sig` sounded weird to me, so I went with `adapted_sig` does that sound alright?